### PR TITLE
(Phase 1) Multi namespace support

### PR DESCRIPTION
**Pull request by jipperinbham**
Addresses https://github.com/compose/transporter/issues/78 based on implementation discussed in https://github.com/compose/transporter/issues/23

The following allows for processing more than 1 collection/table at a time. The current Source adaptors, MongoDB and RethinkDB, have been updated to properly handle this change but the RethinkDB adaptor does NOT automagically pick up new tables added to the database.

An example of how one could pick and choose what namespaces they wanted to sync:

```javascript
pipeline = Source({name:"foofile", namespace: "database./(bas|baz)/"})

pipeline.save({name:"localmongo", namespace: "boom.bas"})
pipeline.save({name:"locales", namespace: "boom.baz"})
```

You can also create transformers in the pipeline which have access to the namespace if you need to alert it inflight.

### CODE REVIEW

**Comment by jipperinbham** [comment]
@alindeman if you can, could you take a look at the RethinkDB changes?

File: pkg/pipe/pipe.go [hunk 1]
```diff
@@ -65,7 +66,7 @@ func NewPipe(pipe *Pipe, path string) *Pipe {
 // Listen starts a listening loop that pulls messages from the In chan, applies fn(msg), a `func(message.Msg) error`, and emits them on the Out channel.
 // Errors will be emited to the Pipe's Err chan, and will terminate the loop.
 // The listening loop can be interupted by calls to Stop().
-func (m *Pipe) Listen(fn func(*message.Msg) (*message.Msg, error)) error {
+func (m *Pipe) Listen(nsFilter *regexp.Regexp, fn func(*message.Msg) (*message.Msg, error)) error {
```
Comment (nstott) [inline pkg/pipe/pipe.go]: i probably would flip the args here..
p.Listen(fn, rx)
(sorry)

File: pkg/adaptor/adaptor.go [hunk 2]
```diff
@@ -108,6 +108,18 @@ func (c *Config) splitNamespace() (string, string, error) {
 	return fields[0], fields[1], nil
 }
 
+// compileNamespace split's on the first '.' and then compiles the second portion to use as the msg filter
+func (c *Config) compileNamespace() (string, *regexp.Regexp, error) {
```
Comment (nstott) [inline pkg/adaptor/adaptor.go]: would it make things cleaner to have a 
`type Namespace regexp.Regexp`
and then have methods `func NewNamespace(s string)`, and `func (ns *Namespace) Match(s) bool `

**Comment by alindeman** [comment]
LGTM

**Comment by nstott** [comment]
looks good, 
i do think that it would make sense to have a namespace type that serves as a container for all the common operations, it would enforce a bit of consistency in how we handled the regexes

we can save that refactor for later though if you want to merge now

**Comment by jipperinbham** [comment]
I reversed the params to `Listen` but left the Namespace refactor for the next PR.

**Comment by Alino** [comment]
I am trying to use this multi namespace but i got some error:

**application.js**
```javascript
pipeline = Source({name:"localmongo", namespace:"admin./(eshops|masterProducts|masterCategories)/"})
pipeline.save({name:"es", namespace:"admin.eshops"})
pipeline.save({name:"es", namespace:"admin.masterProducts"})
pipeline.save({name:"es", namespace:"admin.masterCategories"})
```

**the error:**
Mongo Config {URI:mongodb://192.168.59.103/admin:27017 Namespace:admin./(eshops|masterProducts|masterCategories)/ Ssl:<nil> Timeout: Debug:true Tail:false Wc:0 FSync:false Bulk:false}
setting start timestamp: 6176528679523319808
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [24])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [27])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [19])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [23])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [25])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [22])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [28])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [22])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [26])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [24])
transporter: CRITICAL: elasticsearch error (Bulk Insertion Error. Failed item count [6])

**Comment by jipperinbham** [comment]
@Alino I've just pushed what I believe is a fix for this, c6ab00787620d8211148510f9a5293e566455f14

Would you mind testing on your end?

**Comment by Alino** [comment]
@jipperinbham I can confirm that your fix works, thank you very much!

**Pull request merged**